pandas/core/indexes/range.py

```diff
@@ -1016,6 +1017,22 @@ def _concat(self, indexes: list[Index], name: Hashable) -> Index:
         # In this case return an empty range index.
         return RangeIndex(_empty_range, name=name)
 
+    def _wrap_join_result(
+        self,
+        joined,
+        other,
+        lidx: npt.NDArray[np.intp] | None,
+        ridx: npt.NDArray[np.intp] | None,
+        how: JoinHow,
+    ) -> tuple[Self, npt.NDArray[np.intp] | None, npt.NDArray[np.intp] | None]:
+        join_index, lidx, ridx = super()._wrap_join_result(
+            joined, other, lidx, ridx, how
+        )
+        maybe_ri = self._shallow_copy(join_index._values)
```

Comment:
Can we do this more efficiently based on the inputs?

Reply:
That's fair. Yeah I think working on implementing `RangeIndex._join_monotinic` would be worth it so I'll close this one out